tldw_Server_API/app/core/LLM_Calls/routing/runtime.py

```
@@ -0,0 +1,196 @@
+                "Choose the best model candidate for the request. "
+                "Return only JSON with keys provider and model."
+            ),
+        },
+        {
+            "role": "user",
+            "content": json.dumps(prompt_payload, separators=(",", ":"), ensure_ascii=True),
+        },
+    ]
+
+
+def extract_router_usage(response: Any) -> dict[str, int]:
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `response`

(ANN401)